src/integrity-checker.js

```diff
@@ -171,16 +238,27 @@ export default class InstallationIntegrityChecker {
     }
 
     const linkedModules = this.config.linkedModules;
+
     if (linkedModules.length) {
       result.linkedModules = linkedModules.sort(sortAlpha);
     }
 
-    Object.keys(lockfile).forEach(key => {
+    for (const key of Object.keys(lockfile)) {
       result.lockfileEntries[key] = lockfile[key].resolved || '';
-    });
+    }
+
+    for (const modulesFolder of this._getModulesFolders({workspaceLayout})) {
+      if (await fs.exists(modulesFolder)) {
```

Comment:
I'm still not entirely getting how we may have missing folders in workspaces. Can you provide more details about what this is protecting us from?